framework/src/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -894,7 +892,11 @@ public AnnotationFormatter getAnnotationFormatter() {
      *         empty set if no qualifiers are supported
      */
     public final Set<Class<? extends Annotation>> getSupportedTypeQualifiers() {
-        return supportedQuals;
+        if (this.supportedQuals.isEmpty()) {
+            supportedQuals.addAll(createSupportedTypeQualifiers());
```

Comment:
I modified the contract because supportedQuals` will immediately becomes immutable after calling `createSupportedQualifiers()`. This is ensured by the lazy initialization in `getSupportedQualifiers()`. Thus, required `createSupportedQualifiers()` also return an immutable set seems duplicate.